src/blob/file.rs

```diff
@@ -69,6 +69,44 @@ impl File {
         self.write_at_aio(offset, buf, ioring).await
     }
 
+    pub(crate) async fn write_append_buffers(&self, buffers: Vec<&[u8]>) -> IOResult<usize> {
+        let len = buffers.iter().fold(0, |s, n| s + n.len());
+        if let Some(ref ioring) = self.ioring {
+            self.write_append_buffers_aio(buffers, ioring, len).await?;
+        } else {
+            self.write_append_buffers_sync(buffers, len).await?;
+        }
+        Ok(len)
+    }
+
+    async fn write_append_buffers_aio(
+        &self,
+        buffers: Vec<&[u8]>,
+        ioring: &Rio,
+        len: usize,
+    ) -> IOResult<()> {
+        let mut offset = self.size.fetch_add(len as u64, Ordering::SeqCst);
+        for buf in buffers {
+            self.write_at_aio(offset, buf, ioring).await?;
+            offset = offset + buf.len() as u64;
+        }
+        Ok(())
+    }
+
+    async fn write_append_buffers_sync(&self, buffers: Vec<&[u8]>, len: usize) -> IOResult<()> {
+        let mut offset = self.size.fetch_add(len as u64, Ordering::SeqCst);
+        let file = self.no_lock_fd.clone();
+        let buffers: Vec<_> = buffers.into_iter().map(|v| v.to_vec()).collect();
```

Comment:
The idea was to avoid copying data as much as possible, but here you're converting a slice to a vector, which implies copying

Reply:
As I see, `write_at_sync` is also copy data into vector. This also should be changed.
If there is some issue with passing of ownership, then probably we can use `Bytes` (can be done in #213)

Reply:
It can'be passed without copy indeed, since we need to pass ownership to spawn_blocking. Should I merge this branch into #213 or wait until it's merged into master?

Reply:
I think, it is better to wait until this branch will be merged

Reply:
Or we can do it in reverse: merge #213 to master, because it is ready, and then merge master to the current branch and update to use `Bytes` here

Reply:
I've merged #213

Reply:
Done